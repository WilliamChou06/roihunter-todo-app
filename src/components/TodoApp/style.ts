import styled from 'styled-components';
import {Card, Row} from 'antd';


export const StyledCard = styled(Card)`
border-radius: 6px;
box-shadow: 0 2px 4px rgba(0, 0, 0, 0.18);
overflow: hidden;
min-height: 360px;
min-width: 340px;
max-width: 340px;

.ant-card-body {
  padding: 0;
}

.ant-card-head-title {
  text-align: center;
  font-size: 18px;
}

.ant-spin-container {
  height: 240px;
  overflow-y: auto;
  overflow-x: hidden;

  .ant-list-item {
    word-break: break-all;
    padding-left: 12px;
    padding-right: 12px;
  }
}
`;

export const StyledForm = styled.form`
padding-left: 12px;
padding-right: 24px;
`;

export const AppWrapper = styled(Row)`
height: 100vh;
align-content: center;
`;
