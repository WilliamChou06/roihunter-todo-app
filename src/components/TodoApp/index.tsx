import React, { Component, Suspense, lazy } from 'react';
import { Button, Row, Col, Input, List } from 'antd';
import { StyledCard, StyledForm, AppWrapper } from './style';
import { connect } from 'react-redux';
import {
  startAddTodo,
  startGetSession,
  startSetTodos,
  startDeleteSession
} from '../../actions';
import { Todo } from '../../types';

const TodoItem = lazy(() => import('../TodoItem'));

interface Props {
  startGetSession(): any;
  startDeleteSession(): any;
  startSetTodos(): any;
  startAddTodos(): any;
  startAddTodo(todo: object): any;
  todos: Todo[]

}
interface State {
  todoInput: String;
  todos: Array<Object>;
}

class App extends Component<Props, State> {
  state = {
    todoInput: '',
    todos: []
  };

  handleChange = e => {
    const todoInput = e.target.value;
    this.setState({ todoInput });
  };

  onSubmit = e => {
    e.preventDefault();

    this.props.startAddTodo({ text: this.state.todoInput, urgency: 5 });
    this.setState({ todoInput: '' });
    console.log(this.state.todos);
  };

  renderTodos() {
    this.state.todos.map(todo => todo);
  }

  handleGetSession = () => {
    this.props.startGetSession();
  };

  handleDeleteSession = () => {
    this.props.startDeleteSession();
  };

  componentDidMount() {
    this.props.startSetTodos();
  }

  render() {
    return (
      <AppWrapper type="flex" justify="center">
        <StyledCard title="Todo List">
          <Button onClick={this.handleGetSession}>New session</Button>
          <Button onClick={this.handleDeleteSession}>Delete session</Button>
          <List
            dataSource={this.props.todos}
            renderItem={item => (
              <Suspense fallback={<p>Loading...</p>}>
                <TodoItem todoId={item.id} isCompleted={item.isCompleted}>
                  {item.text}
                </TodoItem>
              </Suspense>
            )}
          />
          <br />
          <StyledForm onSubmit={this.onSubmit}>
            <Row type="flex" justify="space-between">
              <Col span={17}>
                <Input
                  name="todoTitle"
                  type="text"
                  placeholder="Add a todo!"
                  onChange={this.handleChange}
                  value={this.state.todoInput}
                />
              </Col>
              <Col span={6}>
                <Button htmlType="submit" type="primary" ghost>
                  Add Todo
                </Button>
              </Col>
            </Row>
          </StyledForm>
        </StyledCard>
      </AppWrapper>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddTodo: todo => dispatch(startAddTodo(todo)),
  startGetSession: () => dispatch(startGetSession()),
  startSetTodos: () => dispatch(startSetTodos()),
  startDeleteSession: () => dispatch(startDeleteSession())
});

const mapStateToProps = state => {
  return { todos: state.todos };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
