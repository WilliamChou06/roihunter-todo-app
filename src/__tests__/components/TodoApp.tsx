import React from 'react';
import { shallow } from 'enzyme';
import TodoApp from '../../components/TodoApp';

describe('snapshot test', () => {
  it('should render TodoApp component correctly', () => {
    const wrapper = shallow(<TodoApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
