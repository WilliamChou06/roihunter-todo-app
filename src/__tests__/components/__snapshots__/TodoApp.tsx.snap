// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test should render TodoApp component correctly 1`] = `ShallowWrapper {}`;
